php5/plan.sh

```diff
@@ -10,30 +10,68 @@ pkg_source=https://php.net/get/${pkg_distname}-${pkg_version}.tar.bz2/from/this/
 pkg_filename=${pkg_distname}-${pkg_version}.tar.bz2
 pkg_dirname=${pkg_distname}-${pkg_version}
 pkg_shasum=facd280896d277e6f7084b60839e693d4db68318bfc92085d3dc0251fd3558c7
-pkg_deps=(core/libxml2)
-pkg_build_deps=(core/bison2 core/gcc core/make core/re2c)
-pkg_bin_dirs=(bin)
+pkg_deps=(
+  core/coreutils
+  core/curl
+  core/glibc
+  core/libxml2
+  core/openssl
+  core/zlib
+)
+pkg_build_deps=(
+  core/bison2
+  core/gcc
+  core/make
+  core/re2c
+)
+pkg_bin_dirs=(bin sbin)
 pkg_lib_dirs=(lib)
 pkg_include_dirs=(include)
 pkg_interpreters=(bin/php)
 
+do_download() {
+  do_default_download
+  # Download Composer
+  download_file https://getcomposer.org/installer composer-setup.php
+}
+
 do_prepare() {
-  # The configure script expects libxml2 binaries to either be in `/usr/bin`, `/usr/local/bin` or be
-  # passed in as a configure param. Instead of overriding the entire do_build, symlink the
-  # required executable into place.
-  if [[ ! -r /usr/bin/xml2-config ]]; then
-    ln -sv "$(pkg_path_for libxml2)/bin/xml2-config" /usr/bin/xml2-config
-    _clean_xml2=true
-  fi
+  # configure can't locate xml2-config without this
+  PHP_LIBXML_DIR="$(pkg_path_for libxml2)"
+  build_line "Setting PHP_LIBXML_DIR=$PHP_LIBXML_DIR"
+  export PHP_LIBXML_DIR
 }
 
-do_check() {
-  make test
+do_build() {
+  ./configure --prefix="$pkg_prefix" \
+    --enable-exif \
+    --enable-fpm \
+    --enable-mbstring \
+    --enable-opcache \
+    --with-curl="$(pkg_path_for curl)" \
+    --with-openssl="$(pkg_path_for openssl)" \
+    --with-xmlrpc \
+    --with-zlib="$(pkg_path_for zlib)"
+  make
+}
+
+do_install() {
+  do_default_install
+
+  # Install Composer
+  "$pkg_prefix/bin/php" "$HAB_CACHE_SRC_PATH/composer-setup.php" \
+    --filename=composer \
+    --install-dir="$pkg_prefix/bin"
+  fix_interpreter "$pkg_prefix/bin/composer" core/coreutils bin/env
```

Comment:
Another move might be building composer from source and patching this file before it gets built+signed: https://github.com/composer/composer/blob/master/bin/composer